packages/ui/src/components/DatePicker/SDatePickerTableDate.vue

```diff
@@ -0,0 +1,424 @@
+<script setup lang="ts">
+import { getStartDateInCalendar, nextDate } from './date-util'
+import { getDaysInMonth, subMonths, startOfDay, isSameDay } from 'date-fns'
+
+import * as types from './types'
+import { ComputedRef } from 'vue'
+import { daysNames } from './consts'
+import { DatePickerApi, useDatePickerApi } from './api'
+
+const getDateTimestamp = (time: Date | number) => {
+  return startOfDay(time).getTime()
+}
+
+const getFirstDayOfMonth = (date: Date) => {
+  const temp = new Date(date.getTime())
+  temp.setDate(1)
+  return temp.getDay()
+}
+/* eslint-disable complexity */
+
+interface Props {
+  firstDayOfWeek?: number
+  value: Date | Date[]
+  showState: types.ShowState
+  stateStore: types.StateStore
+  hoveredDate: Date
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  firstDayOfWeek: 1,
+  value: () => new Date(),
+})
+
+const state: DatePickerApi = useDatePickerApi()
+const { type, time } = state
+
+const emit = defineEmits(['pick', 'updateShowedState', 'updateHoveredDate'])
+
+const offsetDay = computed(() => {
+  const week = props.firstDayOfWeek
+  return week > 3 ? 7 - week : -week
+})
+const weeks = computed(() => {
+  const week = props.firstDayOfWeek
+  return daysNames.concat(daysNames).slice(week, week + 7)
+})
+const year = computed(() => {
+  return props.showState.year
+})
+const month = computed(() => {
+  return props.showState.month
+})
+const startDateInCalendar = computed(() => {
+  return getStartDateInCalendar(year.value, month.value)
+})
+
+const timeDecoder = (num: number) => {
+  return num.toString().length < 2 ? `0${num}` : num
+}
+
+const getDateTime = (date: Date) => {
+  const hours = timeDecoder(date.getHours())
+  const minutes = timeDecoder(date.getMinutes())
+  return `${hours}:${minutes}`
+}
+
+const dateTableCells: ComputedRef<types.DateTableCell[]> = computed(() => {
```

Comment:
```suggestion
const dateTableCells = computed((): types.DateTableCell[] => {
```

Reply:
This computed seems quite complex and a good candidate to:

- move it out from the component
- cover it with unit tests